lib/elixirls_just_want_to_have_puns.ex

```diff
@@ -28,15 +28,51 @@ defmodule ElixirlsJustWantToHavePuns do
   import Enum
 
   def main([word]) do
-    each(run(word), &(IO.puts(&1)))
+    run(word)
   end
 
   def run(word) do
-    RhymebrainResults.for(word)
-    |> flat_map(&(puns(word, &1)))
+    rhymebrain_results = RhymebrainResults.for(word)
+
+    rhymebrain_results
+    |> Enum.map fn(rhymebrain_result) ->
+      PunMaker.spawn_make(self, word, rhymebrain_result)
+    end
+
+    receive_puns(length(rhymebrain_results), [])
+  end
+
+  defp receive_puns(length, result) do
+    receive do
+      puns ->
+        result = puns ++ result
+
+        if length == 1 do
```

Comment:
Could you define `receive_puns`, with two different implementations and have where that does this when `length==1`? That way this always just called `receive_puns`?